full-service/src/db/encryption_indicator.rs

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) 2020 MobileCoin Inc.
+
+//! DB impl for the Locked model.
+//!
+//! There are two "locked" states for a DB, in order to support a mode in which you can
+//! still submit transactions without accounts (and therefore without a password).
+
+use crate::{
+    db::{
+        account::AccountModel,
+        models::{Account, EncryptionIndicator, NewEncryptionIndicator},
+    },
+    error::WalletDbError,
+};
+
+use diesel::{
+    prelude::*,
+    r2d2::{ConnectionManager, PooledConnection},
+    QueryDsl, RunQueryDsl,
+};
+
+#[derive(Debug)]
+pub enum EncryptionState {
+    /// Database has never been locked, and set_password should be called.
+    Empty,
+    /// Database is encrypted.
+    Encrypted,
+    /// Database is unencrypted. This is an invalid state where some accounts are not encrypted.
```

Comment:
When would the database be in the `Unencrypted` state, if that is an invalid state?